tests/utils/exec/rosacli/ocm_resource_service.go

```diff
@@ -299,6 +306,30 @@ func (arl AccountRoleList) AccountRoles(prefix string) (accountRoles []*AccountR
 	return
 }
 
+// Get all specified account roles by prefix and classic
+func (arl AccountRoleList) DigAccountRoles(prefix string, hcp bool) *AccountRolesUnit {
+	var accRoles *AccountRolesUnit = new(AccountRolesUnit)
+	roleMap := map[string]interface{}{}
+	matchedAccountRoles := arl.AccountRoles(prefix)
+	for _, role := range matchedAccountRoles {
+		if hcp {
+			if strings.Contains(role.RoleName, "HCP") {
+				roleMap[role.RoleType] = role.RoleArn
+				continue
+			}
+
+		} else {
+			if !strings.Contains(role.RoleName, "HCP") {
+				roleMap[role.RoleType] = role.RoleArn
+				continue
+			}
+		}
+
+	}
+	MapStructure(roleMap, accRoles)
+	return accRoles
+}
```

Comment:
I think we can probably flatten that out a little bit
```suggestion
func (arl AccountRoleList) DigAccountRoles(prefix string, hcp bool) *AccountRolesUnit {
	var accRoles *AccountRolesUnit = new(AccountRolesUnit)
	roleMap := map[string]interface{}{}
	matchedAccountRoles := arl.AccountRoles(prefix)
	for _, role := range matchedAccountRoles {
		if hcp && strings.Contains(role.RoleName, "HCP") {
			roleMap[role.RoleType] = role.RoleArn
			continue
		}
		if !strings.Contains(role.RoleName, "HCP") {
			roleMap[role.RoleType] = role.RoleArn
			continue
		}
	}
	MapStructure(roleMap, accRoles)
	return accRoles
}
```